src/TodoApp.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useContext, useState } from 'react';
+import cn from 'classnames';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodosContext } from './contexts/TodosContext';
+import { getRandomId } from './services/todo';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+
+export const App: React.FC = () => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const [todoTitle, setTodoTitle] = useState('');
+
+  const isAllChecked = todos.filter(todo => todo.completed).length
+    === todos.length;
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (todoTitle.trim()) {
```

Comment:
The check for `todoTitle.trim()` is redundant in the if statement. You can directly trim the `todoTitle` before checking.